app/src/main/java/com/github/polypoly/app/ui/menu/lobby/GameLobbyActivity.kt

```diff
@@ -70,162 +83,623 @@ class GameLobbyActivity : ComponentActivity() {
+                    color = MaterialTheme.colors.primary,
+                    fontWeight = FontWeight.Bold
+                ),
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_more_players_needed_count")
+            )
+            Text(
+                text = getString(R.string.game_lobby_more_players_suffix),
+                style = MaterialTheme.typography.body1,
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_more_players_needed_suffix")
+            )
+        }
+    }
+
+    /**
+     * Display the current player count in relation to the maximum player count.
+     * @param players the list of players in the game lobby.
+     * @param maxPlayers the maximum number of players allowed in the game lobby.
+     */
+    @Composable
+    fun CurrentPlayerCountDisplay(players: List<User>, maxPlayers: Int) {
+        Row {
+            Text(
+                text = getString(R.string.game_lobby_players_title),
+                style = MaterialTheme.typography.body1,
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_current_player_count_title")
+            )
+            Text(
+                text =  getString(R.string.game_lobby_players_count, players.size.toString(), maxPlayers.toString()),
+                style = MaterialTheme.typography.body1.copy(
+                    color = MaterialTheme.colors.primary
+                ),
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_current_player_count")
+            )
+        }
+    }
+
+    /**
+     * Launch the game activity and finish this activity. Only available to the admin.
+     * will set the game lobby as started in the database.
+     */
     private fun launchGameActivity(packageContext: Context) {
         val completedLobby = gameLobbyWaitingModel.getGameLobby().value
         if (completedLobby != null) {
-            GameRepository.game = Game.launchFromPendingGame(completedLobby)
-            val gameIntent = Intent(packageContext, GameActivity::class.java)
-            startActivity(gameIntent)
-            finish()
+
+            val newGameLobby = GameLobby(
+                completedLobby.admin,
+                completedLobby.rules,
+                completedLobby.name,
+                completedLobby.code,
+                completedLobby.private,
+                true
+            )
+            remoteDB.updateValue(newGameLobby)
+
+            navigateToGame(packageContext, newGameLobby)
         }
     }
 
-    @Composable
-    private fun QuitButton() {
-        Button(onClick = { /*TODO*/ }) {
-            Text(text = "QUIT")
+    /**
+     * Leave the lobby and return to the main menu.
+     * Makes sure that the admin is replaced by another user if the admin leaves.
+     */
+    private fun leaveLobby(gameLobby: GameLobby) {
```

Comment:
This logic should be moved to `gameLobbyWaitingModel` instead of being kept in the UI part

Reply:
I created an issue for this comment and the one above also related to business logic: https://github.com/polypoly-team/android-app/issues/165